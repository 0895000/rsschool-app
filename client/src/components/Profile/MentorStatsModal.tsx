import * as React from 'react';
import { MentorStats } from '../../../../common/models/profile';
import {
  Modal,
  Avatar,
  Typography,
  Tag,
  Row,
  Col,
} from 'antd';

const { Text } = Typography;

import {
  GithubOutlined,
  LockFilled,
} from '@ant-design/icons';

type Props = {
  stats: MentorStats;
  isVisible: boolean;
  onHide: () => void;
};

class MentorStatsModal extends React.Component<Props> {
  render() {
    const { stats, isVisible, onHide } = this.props;
    const { courseFullName, students, courseName } = stats;
    const courseYearPostfix = courseName.replace(/ /gi, '');

    return (
      <Modal
        title={`${courseFullName} statistics`}
        visible={isVisible}
        onCancel={onHide}
        footer={null}
        width={'80%'}
      >
        <Row gutter={[16, 16]}>
        {
          students.map(({ name, githubId, isExpelled, totalScore }) => {
            const profile = `/profile?githubId=${githubId}`;
            const guithubLink = `https://github.com/${githubId}`;
            const privateRepoLink = `https://github.com/rolling-scopes-school/${githubId}-${courseYearPostfix}`;

            return (
              <Col
                key={`mentor-stats-modal-student-${githubId}`}
                xs={{span: 24}}
                sm={{span: 12}}
                md={{span: 8}}
                lg={{span: 6}}
              >
                <Row style={{ marginBottom: 24 }}>
                  <Col>
                    <Avatar
                      src={`${guithubLink}.png?size=${128}`}
                      alt={`${githubId} avatar`}
                      size={64}
                      style={{ marginRight: 24 }}
                    />
                  </Col>
                  <Col>
                    <p style={{ fontSize: 16 }}>
                      <Text strong>
                        <a href={profile}>
                          {name}
                        </a>
                      </Text>
                    </p>
                    <p style={{ marginBottom: 5 }}>
                    {
                      isExpelled ?
                        <Tag color="red">expelled</Tag> :
                        <Tag color="green">active</Tag>
                    }
                    </p>
                    <p style={{ marginBottom: 5 }}>
                      Score:  <Text mark>{totalScore}</Text>
                    </p>
                    <p style={{ marginBottom: 5 }}>
                      <GithubOutlined /> <a href={guithubLink} target="_blank">{githubId}</a>
                    </p>
                    <p style={{ marginBottom: 5 }}>
                      <LockFilled /> <a href={privateRepoLink} target="_blank">
                        {`${githubId}-${courseYearPostfix}`}
                      </a>
                    </p>
                  </Col>
                </Row>
              </Col>
            )
          })
        }
        </Row>
      </Modal>
    );
  }
}

export default MentorStatsModal;
