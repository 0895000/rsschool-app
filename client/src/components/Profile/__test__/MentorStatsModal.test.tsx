import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import MentorStatsModal from '../MentorStatsModal';

describe('MentorStatsModal', () => {
  it('Should render correctly', () => {
    const output = shallow(
      <MentorStatsModal
        stats={{
          courseName: 'rs-2018-q1',
          locationName: 'Minsk',
          courseFullName: 'Rolling Scopes School 2018-Q1',
          students: [
            {
              githubId: 'alex',
              name: 'Alex Petrov',
              isExpelled: false,
              totalScore: 3453,
            },
            {
              githubId: 'vasya',
              name: 'Vasiliy Alexandrov',
              isExpelled: true,
              totalScore: 120,
            },
          ],
        }}
        isVisible={true}
        onHide={jest.fn()}
      />,
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
