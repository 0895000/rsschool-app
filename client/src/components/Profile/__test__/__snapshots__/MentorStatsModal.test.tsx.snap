// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MentorStatsModal Should render correctly 1`] = `
<Modal
  confirmLoading={false}
  footer={null}
  maskTransitionName="fade"
  okType="primary"
  onCancel={[Function]}
  title="Rolling Scopes School 2018-Q1 statistics"
  transitionName="zoom"
  visible={true}
  width="80%"
>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col
      key="mentor-stats-modal-student-alex"
      lg={
        Object {
          "span": 6,
        }
      }
      md={
        Object {
          "span": 8,
        }
      }
      sm={
        Object {
          "span": 12,
        }
      }
      xs={
        Object {
          "span": 24,
        }
      }
    >
      <Row
        gutter={0}
        style={
          Object {
            "marginBottom": 24,
          }
        }
      >
        <Col>
          <Avatar
            alt="alex avatar"
            shape="circle"
            size={64}
            src="https://github.com/alex.png?size=128"
            style={
              Object {
                "marginRight": 24,
              }
            }
          />
        </Col>
        <Col>
          <p
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            <Text
              strong={true}
            >
              <a
                href="/profile?githubId=alex"
              >
                Alex Petrov
              </a>
            </Text>
          </p>
          <p
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            <Tag
              closable={false}
              color="green"
            >
              active
            </Tag>
          </p>
          <p
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            Score:  
            <Text
              mark={true}
            >
              3453
            </Text>
          </p>
          <p
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            <ForwardRef(GithubOutlined) />
             
            <a
              href="https://github.com/alex"
              target="_blank"
            >
              alex
            </a>
          </p>
          <p
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            <ForwardRef(LockFilled) />
             
            <a
              href="https://github.com/rolling-scopes-school/alex-rs-2018-q1"
              target="_blank"
            >
              alex-rs-2018-q1
            </a>
          </p>
        </Col>
      </Row>
    </Col>
    <Col
      key="mentor-stats-modal-student-vasya"
      lg={
        Object {
          "span": 6,
        }
      }
      md={
        Object {
          "span": 8,
        }
      }
      sm={
        Object {
          "span": 12,
        }
      }
      xs={
        Object {
          "span": 24,
        }
      }
    >
      <Row
        gutter={0}
        style={
          Object {
            "marginBottom": 24,
          }
        }
      >
        <Col>
          <Avatar
            alt="vasya avatar"
            shape="circle"
            size={64}
            src="https://github.com/vasya.png?size=128"
            style={
              Object {
                "marginRight": 24,
              }
            }
          />
        </Col>
        <Col>
          <p
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            <Text
              strong={true}
            >
              <a
                href="/profile?githubId=vasya"
              >
                Vasiliy Alexandrov
              </a>
            </Text>
          </p>
          <p
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            <Tag
              closable={false}
              color="red"
            >
              expelled
            </Tag>
          </p>
          <p
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            Score:  
            <Text
              mark={true}
            >
              120
            </Text>
          </p>
          <p
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            <ForwardRef(GithubOutlined) />
             
            <a
              href="https://github.com/vasya"
              target="_blank"
            >
              vasya
            </a>
          </p>
          <p
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            <ForwardRef(LockFilled) />
             
            <a
              href="https://github.com/rolling-scopes-school/vasya-rs-2018-q1"
              target="_blank"
            >
              vasya-rs-2018-q1
            </a>
          </p>
        </Col>
      </Row>
    </Col>
  </Row>
</Modal>
`;
